yt/geometry/particle_oct_container.pyx

```diff
@@ -746,9 +746,9 @@ cdef class ParticleBitmap:
         # Loop over positions skipping those outside the domain
         cdef np.ndarray[np.uint64_t, ndim=1, cast=True] sorted_order
         if hsml is None:
-            sorted_order = np.argsort(morton_indices)
+            sorted_order = np.argsort(morton_indices).astype(np.uint64, copy=False)
```

Comment:
One last thing, could you add this comment here as well as in the other affected file ?
```suggestion
            # casting to uint64 for compatibility with 32 bits systems
            # see https://github.com/yt-project/yt/issues/3656
            sorted_order = np.argsort(morton_indices).astype(np.uint64, copy=False)
```

Reply:
Sure; pushed.